tools/scylla-nodetool.cc

```diff
@@ -1336,6 +1336,15 @@ void snapshot_operation(scylla_rest_client& client, const bpo::variables_map& vm
 
     sstring kn_msg;
 
+    auto split_kt = [] (const std::string_view kt) -> std::optional<std::pair<sstring, sstring>> {
+        std::vector<sstring> components;
+        boost::split(components, kt, boost::algorithm::is_any_of("."));
+        if (components.size() == 2) {
+            return std::pair(components[0], components[1]);
+        }
+        return {};
+    };
```

Comment:
Alternator tables and keyspaces may have a dot in their name, so the ks.tbl convention isn't good enough. Luckily most tools take the keyspace and table argument separately but as you noted some take them together.
So in https://github.com/scylladb/scylla-tools-java/commit/0187829d5eeaabd3c0b48887923dc029232f62d4 I introduced to the Java nodetool a new convention that allows "ks/tbl" and also "ks.name/" to refer to a keyspace (not a table) that has a dot in its name).
Please allow this convention in the new Java-free nodetool as well, so we can work on Alternator tables too.

Reply:
I checked https://github.com/scylladb/scylla-tools-java/commit/0187829d5eeaabd3c0b48887923dc029232f62d4 and it doesn't seem to modify `nodetool snapshot`. Note that snapshot allows for providing keyspace and tables separately, although `ks.name/` would probably be mis-handled by the current code.
I'll send a v2 with support for the above semantics.